Simplified/NYPLUserAccount.swift

```diff
@@ -36,47 +135,85 @@ extension Notification.Name {
   {
     shared.accountInfoLock.lock()
     defer {
-        shared.accountInfoLock.unlock()
+      shared.accountInfoLock.unlock()
     }
     if let uuid = libraryUUID,
-        uuid != AccountsManager.NYPLAccountUUIDs[0]
+      uuid != AccountsManager.NYPLAccountUUIDs[0]
     {
-      shared.barcodeKey = "NYPLAccountBarcode_\(uuid)"
-      shared.authorizationIdentifierKey = "NYPLAccountAuthorization_\(uuid)"
-      shared.PINKey = "NYPLAccountPIN_\(uuid)"
-      shared.adobeTokenKey = "NYPLAccountAdobeTokenKey_\(uuid)"
-      shared.patronKey = "NYPLAccountPatronKey_\(uuid)"
-      shared.authTokenKey = "NYPLAccountAuthTokenKey_\(uuid)"
-      shared.adobeVendorKey = "NYPLAccountAdobeVendorKey_\(uuid)"
-      shared.providerKey = "NYPLAccountProviderKey_\(uuid)"
-      shared.userIDKey = "NYPLAccountUserIDKey_\(uuid)"
-      shared.deviceIDKey = "NYPLAccountDeviceIDKey_\(uuid)"
-      shared.licensorKey = "NYPLAccountLicensorKey_\(uuid)"
+      shared.libraryUUID = uuid
     } else {
-      shared.barcodeKey = "NYPLAccountBarcode"
-      shared.authorizationIdentifierKey = "NYPLAccountAuthorization"
-      shared.PINKey = "NYPLAccountPIN"
-      shared.adobeTokenKey = "NYPLAccountAdobeTokenKey"
-      shared.patronKey = "NYPLAccountPatronKey"
-      shared.authTokenKey = "NYPLAccountAuthTokenKey"
-      shared.adobeVendorKey = "NYPLAccountAdobeVendorKey"
-      shared.providerKey = "NYPLAccountProviderKey"
-      shared.userIDKey = "NYPLAccountUserIDKey"
-      shared.deviceIDKey = "NYPLAccountDeviceIDKey"
-      shared.licensorKey = "NYPLAccountLicensorKey"
+      shared.libraryUUID = nil
     }
-    
+
     return shared
   }
-   
+
+  private func notifyAccountDidChange() {
+    NotificationCenter.default.post(
+      name: Notification.Name.NYPLUserAccountDidChange,
+      object: self
+    )
+  }
+
+  // MARK: - Storage
+  private lazy var _authorizationIdentifier: KeychainVariable<String> = StorageKey.authorizationIdentifier
+    .keyForLibrary(uuid: libraryUUID)
+    .asKeychainVariable(with: accountInfoLock)
+  private lazy var _adobeToken: KeychainVariable<String> = StorageKey.adobeToken
```

Comment:
did you test this is backward compatible once a user upgrades to this code? This set of changes would require a meticulous QA effort. @kyles_ep knows more about this area of the code.

Reply:
Besides token, pin and barcode, other variables SHOULD work on par with old logic, I'll do more tests to verify it